### PR TITLE
Main <- Dev

### PR DESCRIPTION
**Pull request by hmrtn**
Adds zkKYC demo repo

### CODE REVIEW

File: zk-kyc/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# zk-KYC Demonstration Application
+
```
Comment (intoverflow) [inline zk-kyc/README.md:2]: Needs a disclaimer about the demo being experimental
```suggestion
> WARNING: this project is experimental and should not be used for any production use cases. Use at your own risk.
```

**Review by intoverflow — CHANGES_REQUESTED** [review]
We're missing the license/copyright notice on several files. Otherwise, looks good. I'll take another pass through this afternoon or tomorrow.

**Comment by hmrtn** [comment]
@intoverflow I added the standard license header to all the rust files. How should we license the solidity files and react application?

**Comment by intoverflow** [comment]
> @intoverflow I added the standard license header to all the rust files. How should we license the solidity files and react application?

Let's do them both with Apache 2. Here's an example from the verifier contract:

https://github.com/risc0/risc0/blob/main/bonsai/ethereum/contracts/IRiscZeroVerifier.sol

File: zk-kyc/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# zk-KYC Demonstration Application
+> WARNING: this project is experimental and should not be used for any production use cases. Use at your own risk.
```
Comment (intoverflow) [inline zk-kyc/README.md:2]: ✅

File: zk-kyc/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+Deploy the contracts to a local node:
+
+```bash
+forge script script/Deploy.s.sol --rpc-url http://localhost:8545
+```
+
+## Privacy and Security
+
+As a sample application, the ERC721 Token Minting Demo is designed for demonstration purposes and is not suitable for production use. It is important to be aware that:
```
Comment (intoverflow) [inline zk-kyc/README.md:84]: ✅

File: zk-kyc/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# zk-KYC Demonstration Application
+> WARNING: this project is experimental and should not be used for any production use cases. Use at your own risk.
+The zk-KYC demo application allows individuals to verify and mint their identity as an NFT enabling onchain identity verification with zero-knowledge proofs.
+
+This demo leverages ID.me to generate a client authentication token. The token includes a nonce that is associated with the user's connected wallet address, employing principles from the [OpenPubkey: Augmenting OpenID Connect with User held Signing Keys](https://eprint.iacr.org/2023/296) paper. The JWT's integrity is then verified within the guest using ID.me's public RS256 signing [certificates](https://api.id.me/oidc/.well-known/jwks). Subsequently, the guest generates a cryptographic proof of the JWT's integrity and issues a receipt. This receipt, containing the SNARK, an obfuscated identifier, and the user's address, can be validated on the onchain for ERC721 token minting or other transactions, if valid.
```
Comment (intoverflow) [inline zk-kyc/README.md:5]:
```suggestion
This demo leverages [ID.me](https://www.id.me/) to generate a client authentication token. The token includes a nonce that is associated with the user's connected wallet address, employing principles from the [OpenPubkey: Augmenting OpenID Connect with User held Signing Keys](https://eprint.iacr.org/2023/296) paper. The JWT's integrity is then verified within the guest using ID.me's public RS256 signing [certificates](https://api.id.me/oidc/.well-known/jwks). Subsequently, the guest generates a cryptographic proof of the JWT's integrity and issues a receipt. This receipt, containing the SNARK, an obfuscated identifier, and the user's address, can be validated on the onchain for ERC721 token minting or other transactions, if valid.
```

File: zk-kyc/contracts/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Smart Contracts
+
+This directory hosts a Foundry project dedicated to the smart contracts utilized by the demo application. Please note that these contracts are unaudited and are intended solely for demonstration purposes.
```
Comment (intoverflow) [inline zk-kyc/contracts/README.md:3]: ✅

**Review by intoverflow — APPROVED** [review]
Beautiful 😍

**Pull request merged**